### PR TITLE
ddl: clean ingest temp data dir in background loop

### LINKED ISSUE #54007: Decouple clean up ingest temp data from `BackendCtxMgr`
## Enhancement

Use a separate loop for cleaning up is better. Otherwise, if there are no new add index jobs running, temp-dir cannot be cleaned up.

### PR DESCRIPTION
**Pull request by tangenta**
### What problem does this PR solve?

Issue Number: close #54007

Problem Summary:

### What changed and how does it work?

Clean ingest temp data dir in background loop

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/ddl/ingest/env.go [hunk 1]
```diff
@@ -108,5 +110,82 @@ func genLightningDataDir() (string, error) {
 	return sortPath, nil
 }
 
-// GenLightningDataDirForTest is only used for test.
-var GenLightningDataDirForTest = genLightningDataDir
+// CleanUpTempDir reads the temp dir and removes the stale index data.
+// This function will get
```
Comment (lance6716) [inline pkg/ddl/ingest/env.go:114]: comment is not completed

File: pkg/ddl/ingest/env.go [hunk 2]
```diff
@@ -108,5 +110,82 @@ func genLightningDataDir() (string, error) {
...
+func CleanUpTempDir(ctx context.Context, se sessionctx.Context, path string) {
+	err := os.MkdirAll(path, 0700)
```
Comment (lance6716) [inline pkg/ddl/ingest/env.go:116]: clean up don't need to create the folder 🤔

File: pkg/ddl/ingest/env.go (hunk 1 above)
Comment (tangenta) [reply pkg/ddl/ingest/env.go]: https://github.com/pingcap/tidb/pull/54008/commits/2a3823e2ba1edb02981b313ce5ea1ad78437d694

File: pkg/ddl/ingest/env.go (hunk 2 above)
Comment (tangenta) [reply pkg/ddl/ingest/env.go]: https://github.com/pingcap/tidb/pull/54008/commits/2a3823e2ba1edb02981b313ce5ea1ad78437d694

**Review by wjhuang2016 — APPROVED** [review]

**Pull request merged**